api_app/core/dataclasses.py

```diff
@@ -94,3 +97,14 @@ def _read_secrets(self, secrets_filter=[]) -> dict:
     @abstractmethod
     def get_full_import_path(self) -> str:
         raise NotImplementedError()
+
+    @classmethod
+    def get(cls, name: str) -> typing.Optional["AbstractConfig"]:
```

Comment:
Although I like that we can use an abstract method like this, but because of how things are intertwined at the moment this will cause circular imports and what not.

Reply:
I'd say directly define the `get` function in the subclasses.

Reply:
and in the `AbstractConfig`, wrap it w/ the `abstractmethod` decorator and call `raise NotImplemented()`.